odoo/addons/base/models/res_partner.py

```diff
@@ -268,15 +268,21 @@ def _compute_partner_share(self):
         for partner in self - super_partner:
             partner.partner_share = not partner.user_ids or not any(not user.share for user in partner.user_ids)
 
-    @api.depends('vat')
+    @api.depends('vat', 'company_id')
     def _compute_same_vat_partner_id(self):
         for partner in self:
             # use _origin to deal with onchange()
             partner_id = partner._origin.id
-            domain = [('vat', '=', partner.vat)]
+            #active_test = False because if a partner has been deactivated you still want to raise the error,
+            #so that you can reactivate it instead of creating a new one, which would loose its history.
+            Partner = self.with_context(active_test=False).sudo()
+            domain = [
+                ('vat', '=', partner.vat),
+                ('company_id', 'in', [False, partner.company_id.id]),
+            ]
             if partner_id:
                 domain += [('id', '!=', partner_id), '!', ('id', 'child_of', partner_id)]
```

Comment:
I'm now thinking if this domain is correct, as `child_of` operator should use list instead of an id. Can it be or does it accept both?

Reply:
whatever the default record rule says, I'd say :thinking:

Reply:
I see both forms over whole Odoo code, so I suppose both are valid. Sorry for the noise.

Reply:
You can use both
https://github.com/odoo/odoo/blob/668778125c2d4231d9b48456c3397edb8d04acb3/odoo/addons/base/tests/test_expression.py#L122-L128

Reply:
Thanks for the confirmation :smiley: